### DISCUSSION TITLE
Rspack support?

### DISCUSSION THREAD

**Discussion by Zercerium** [posva/unplugin-vue-router#467] [Questions and Help] [answered]
Hi,

are you planing to support Rspack/Rsbuild?

I tried out to use the webpack plugin after this instruction https://rsbuild.dev/plugins/dev/index#registering-rspack-plugin

but got the following error:
`TypeError: Plugin is not a constructor`

Thanks for your work!

**Comment by posva** [comment, marked as answer]
It's an unplugin so it should be able to support rspack (except for HMR): https://unplugin.unjs.io/guide/#bundler-framework-integration

You can do
```ts
import VueRouter from 'unplugin-vue-router'
// then use the one you need
VueRouter.rspack
VueRouter.vite
// etc.
```

**Reply by Zercerium** [reply]
thanks a lot! it works :)

```ts
import { defineConfig } from '@rsbuild/core';
import { pluginVue } from '@rsbuild/plugin-vue';
import VueRouter from 'unplugin-vue-router';

export default defineConfig({
  plugins: [pluginVue()],
  tools: {
    rspack: {
      plugins: [VueRouter.rspack()],
    },
  },
});
```
